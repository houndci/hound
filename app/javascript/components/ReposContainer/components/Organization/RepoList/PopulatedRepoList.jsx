import React from 'react';

import Repo from './../../Repo';

export default class PopulatedRepoList extends React.Component {
  canShow(repo) {
    const { filterTerm } = this.props;

    if (filterTerm === null) {
      return true;
    } else {
      const repoName = repo.name.toLowerCase();
      return repoName.indexOf(filterTerm.toLowerCase()) !== -1;
    }
  }

  render() {
    const { repos, onRepoClicked, isProcessingId } = this.props;

    return (
      <ul className="repos">
        {repos.filter((repo) => this.canShow(repo)).map( repo => (
          <Repo
            repo={repo}
            key={repo.id}
            onRepoClicked={onRepoClicked}
            isProcessingId={isProcessingId}
          />
        ))}
      </ul>
    );
  }
}
