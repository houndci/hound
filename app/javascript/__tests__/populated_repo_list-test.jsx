import PopulatedRepoList from '../components/ReposContainer/components/Organization/RepoList/PopulatedRepoList';

it('renders a list of repos appropriately', () => {
  const organizations = [
    { id: 1, name: "Test org" }
  ];
  const repos = [
    {
      id: 1,
      name: "Test repo",
      owner: {
        id: 1
      }
    }
  ];

  const wrapper = shallow(
    <PopulatedRepoList
      repos={repos}
      onRepoClicked={jest.fn()}
      isProcessingId={null}
      filterTerm={null}
    />
  );
  expect(wrapper).toMatchSnapshot();

  const wrapper2 = shallow(
    <PopulatedRepoList
      repos={repos}
      onRepoClicked={jest.fn()}
      isProcessingId={null}
      filterTerm={""}
    />
  );
  expect(wrapper2).toMatchSnapshot();
});

it('filters a list of repos appropriately', () => {
  const organizations = [
    { name: "S.H.I.E.L.D." },
    { name: "H.A.M.M.E.R." },
    { name: "Hydra" },
  ];
  const repos = [
    {
      id: 666,
      name: "Hydra/deathray",
    },
    {
      id: 302,
      name: "S.H.I.E.L.D./tesseract",
    },
    {
      id: 302,
      name: "H.A.M.M.E.R./iron_man_suit_knockoff",
    }
  ];

  const wrapper = shallow(
    <PopulatedRepoList
      repos={repos}
      onRepoClicked={jest.fn()}
      isProcessingId={null}
      filterTerm={"Hydra"}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
