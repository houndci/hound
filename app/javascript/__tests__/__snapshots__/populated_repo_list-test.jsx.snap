// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters a list of repos appropriately 1`] = `
<ul
  className="repos"
>
  <Repo
    isProcessingId={null}
    key="666"
    onRepoClicked={[MockFunction]}
    repo={
      Object {
        "id": 666,
        "name": "Hydra/deathray",
      }
    }
  />
</ul>
`;

exports[`renders a list of repos appropriately 1`] = `
<ul
  className="repos"
>
  <Repo
    isProcessingId={null}
    key="1"
    onRepoClicked={[MockFunction]}
    repo={
      Object {
        "id": 1,
        "name": "Test repo",
        "owner": Object {
          "id": 1,
        },
      }
    }
  />
</ul>
`;

exports[`renders a list of repos appropriately 2`] = `
<ul
  className="repos"
>
  <Repo
    isProcessingId={null}
    key="1"
    onRepoClicked={[MockFunction]}
    repo={
      Object {
        "id": 1,
        "name": "Test repo",
        "owner": Object {
          "id": 1,
        },
      }
    }
  />
</ul>
`;
